src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -218,43 +403,58 @@ void extractRecurrence(const QJsonArray &recurrence, KCalCore::Event::Ptr event,
         return;
     }
 
+    kcalRecurrence->clear(); // avoid adding duplicate recurrence information
```

Comment:
Are RDATEs not covered by the recurs() check?  Does that one only cover RRULE?  Should I add clear() inside the if block also, in that case?